jung-algorithms/src/main/java/edu/uci/ics/jung/layout/spatial/Rectangle.java

```diff
@@ -45,11 +53,53 @@ public boolean intersects(Rectangle other) {
     return maxX >= other.x && other.maxX >= x && maxY >= other.y && other.maxY >= y;
   }
 
+  /**
+   * @param p point to test
+   * @return true if the coordinate is contained, false otherwise
+   */
   public boolean contains(Point p) {
     return contains(p.x, p.y);
   }
 
+  /**
+   * Fail fast for the most common case where the point coordinates are not contained.
+   * Implementation leaves space for debug breakpoints
+   *
+   * @param ox coodinate to test
+   * @param oy coordinate to test
+   * @return true if the coordinate is contained, false otherwise
+   */
   public boolean contains(double ox, double oy) {
-    return ox >= this.x && ox <= maxX && oy >= this.y && oy <= maxY;
+    if (ox < x) return false;
+    if (ox > maxX) return false;
+    if (oy < y) return false;
+    if (oy > maxY) return false;
+    return true;
+  }
+
+  public Rectangle add(double newX, double newY) {
```

Comment:
I borrowed the name from the matching method in Rectangle2D. Your (our) confusion about what it does is well illustrated by the fact that I previously wrote my own method to union a new point with an existing Rectangle, because I did not find their 'add' method while looking at the various ways to construct a union (named `createUnion`)